### ISSUE TITLE
lots of "warning: comparison of integer expressions of different signedness"

### ISSUE THREAD

**Issue by vtorri** [lexbor/lexbor#171]
there are a lot of these warnings, mainly because of  comparison of ptrdiff_t value with a size_t value

I don't know if they should be fixed or not.

**Comment by lexborisov** [comment]
@vtorri 

Please, a warming example (file, line).
OS name, compiler?

**Comment by vtorri** [comment]
for example

```
E:/Documents/programmes_x64/msys2/home/vtorri/gitroot_64/lexbor/source/lexbor/dom/interfaces/node.c: In function 'lxb_dom_node_by_class_name_cb':
E:/Documents/programmes_x64/msys2/home/vtorri/gitroot_64/lexbor/source/lexbor/dom/interfaces/node.c:746:45: warning: comparison of integer expressions of different signedness: 'long long int' and 'size_t' {aka 'long long unsigned int'} [-Wsign-compare]
  746 |             if (pos != data && (data - pos) == cb_ctx->value_length) {
      |                                             ^~
E:/Documents/programmes_x64/msys2/home/vtorri/gitroot_64/lexbor/source/lexbor/dom/interfaces/node.c:775:21: warning: comparison of integer expressions of different signedness: 'long long int' and 'size_t' {aka 'long long unsigned int'} [-Wsign-compare]
  775 |     if ((end - pos) == cb_ctx->value_length) {
      |                     ^~
```

it's on windows with gcc 64 bits

**Comment by lexborisov** [comment]
@vtorri 

Thanks for the report!
Fixed in [76bc5ab](https://github.com/lexbor/lexbor/commit/76bc5ab1b2b02bf25334bce1a767f9a211bfaaa9).
